modules/passkey/test/userStories/executeUserOpFromPasskeySigner.ts

```diff
@@ -0,0 +1,147 @@
+      publicKey: {
+        rp: {
+          name: 'Safe',
+          id: 'safe.global',
+        },
+        user: {
+          id: ethers.getBytes(ethers.id('chucknorris')),
+          name: 'chucknorris',
+          displayName: 'Chuck Norris',
+        },
+        challenge: ethers.toBeArray(Date.now()),
+        pubKeyCredParams: [{ type: 'public-key', alg: -7 }],
+      },
+    })
+
+    // Deploy a signer contract
+    const publicKey = decodePublicKey(credential.response)
+    await signerFactory.createSigner(publicKey.x, publicKey.y, await verifier.getAddress())
+    const signer = await signerFactory.getSigner(publicKey.x, publicKey.y, await verifier.getAddress())
+
+    // Deploy a Safe with passkey signer as owner
+
+    // The initializer data to enable the Safe4337Module as a module on a Safe
+    const initializer = safeModuleSetup.interface.encodeFunctionData('enableModules', [[module.target]])
+
+    // Create setup data to deploy a Safe with EOA and passkey signer as owners, threshold 1, Safe4337Module as module and fallback handler
+    const setupData = singleton.interface.encodeFunctionData('setup', [
+      [signer],
+      1n,
+      safeModuleSetup.target,
+      initializer,
+      module.target,
+      ethers.ZeroAddress,
+      0,
+      ethers.ZeroAddress,
+    ])
+
+    // Deploy a Safe with EOA and passkey signer as owners
+    const safeSalt = Date.now()
+    const safeAddress = await proxyFactory.createProxyWithNonce.staticCall(singleton, setupData, safeSalt)
+    await proxyFactory.createProxyWithNonce(singleton, setupData, safeSalt)
+
+    return {
+      relayer,
+      module,
+      entryPoint,
+      navigator,
+      signer,
+      credential,
+      safeAddress,
+    }
+  })
+
+  it('should execute a userOp with WebAuthn signer as owner', async () => {
+    // Step 1: Setup the contracts
+    const { relayer, module, entryPoint, navigator, signer, credential, safeAddress } = await setupTests()
+
+    // Step 2: Create a userOp and sign it using passkey credential.
+    const safeOp = buildSafeUserOpTransaction(
+      safeAddress,
+      ethers.ZeroAddress,
+      ethers.parseEther('0.2'),
+      '0x',
+      await entryPoint.getNonce(safeAddress, 0),
+      await entryPoint.getAddress(),
+      false,
+      true,
+    )
+
+    const packedUserOp = buildPackedUserOperationFromSafeUserOperation({
+      safeOp: safeOp,
+      signature: '0x',
+    })
+
+    const opHash = await module.getOperationHash(packedUserOp)
+
+    const assertion = navigator.credentials.get({
+      publicKey: {
+        challenge: ethers.getBytes(opHash),
+        rpId: 'safe.global',
+        allowCredentials: [{ type: 'public-key', id: new Uint8Array(credential.rawId) }],
+        userVerification: 'required',
+      },
+    })
+
+    // Build contract signature that the Safe will forward to the signer contract
+    const signature2 = buildSignatureBytes([
+      {
+        signer: signer as string,
```

Comment:
why is the casting to string required? isn't it a string already?

Reply:
it was residue from the initial code. Its not required now. removed in https://github.com/safe-global/safe-modules/pull/344/commits/f84c241a2faf9000dbebff38d522d446e42d3d8c